### ISSUE TITLE
[BUG] java.time support goes through java.sql.Timestamp

### ISSUE THREAD

**Issue by marschall** [microsoft/mssql-jdbc#1143]
## Driver version
7.4.1.jre8

## SQL Server version
Microsoft SQL Server 2017 (RTM-CU13) (KB4466404) - 14.0.3048.4 (X64) 
	Nov 30 2018 12:57:58 
	Copyright (C) 2017 Microsoft Corporation
	Developer Edition (64-bit) on Linux (Ubuntu 16.04.5 LTS)

## Client Operating System
Linux hostname 5.0.0-27-generic #28-Ubuntu SMP Tue Aug 20 19:53:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

## JAVA/JVM version
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (Zulu 8.40.0.25-CA-linux64) (build 1.8.0_222-b10)
OpenJDK 64-Bit Server VM (Zulu 8.40.0.25-CA-linux64) (build 25.222-b10, mixed mode)

## Table schema
empty

## Problem description

1. To create a `java.time` type first a `java.sql.Timestamp` is created see
https://github.com/microsoft/mssql-jdbc/blob/8a5fca2207dd9a5f4f22bfbfb45652992a53cb3a/src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java#L2384

This is an issue because `java.sql.Timestamp` is bound to the JVM time zone. If the value does not exist in a JVM time zone because the DST offset changes a wrong value is returned.

## Reproduction code

```java

  @Test
  public void timestampConversion() throws SQLException {
    LocalDateTime unstorableValue = this.getUnstorableValue();
    try (Connection connection = this.dataSource.getConnection();
         PreparedStatement selectStatement = connection.prepareStatement("SELECT '" + unstorableValue.toLocalDate() + " " + unstorableValue.toLocalTime() + "'");
         ResultSet resultSet = selectStatement.executeQuery()) {
      while (resultSet.next()) {
        LocalDateTime actual = resultSet.getObject(1, LocalDateTime.class);
        if (!unstorableValue.equals(actual)) {
          System.out.println("expected: " + unstorableValue + " actual: " + actual);
        }
      }
    }
  }


  LocalDateTime getUnstorableValue() {
    ZoneId systemTimezone = ZoneId.systemDefault();
    Instant now = Instant.now();

    ZoneRules rules = systemTimezone.getRules();
    ZoneOffsetTransition transition = rules.nextTransition(now);
    assertNotNull(transition, "test has to be run in a time zone with DST");
    if (!transition.getDateTimeBefore().isBefore(transition.getDateTimeAfter())) {
      transition = rules.nextTransition(transition.getInstant().plusSeconds(1L));
      assertNotNull(transition, "test has to be run in a time zone with DST");
    }

    Duration gap = Duration.between(transition.getDateTimeBefore(), transition.getDateTimeAfter());
    LocalDateTime unstorable = transition.getDateTimeBefore().plus(gap.dividedBy(2L));
    // make sure none of the nano seconds are 0 so that we can detect truncation
    return unstorable.withNano(123456789).truncatedTo(ChronoUnit.MICROS);
  }

```

**Comment by peterbae** [comment]
Hi @marschall, thanks for the detailed information. The team is aware of the issues regarding the retrieval of column data as LocalDateTime from a ResultSet (this issue is related to #1070 and #1088), and the team is working on a fix. Will let you know once we have an update.

**Comment by peterbae** [comment]
Fixed in #1200.
